pe32me162ir_pub.ino

```diff
@@ -112,15 +74,43 @@ const char mqtt_topic[] = "some/topic";
 # endif
 #endif
 
-/* Load PROGMEM macro to store global constants in program/flash memory.
- * We define the AS_MEM_CSTR and AS_PGM_CSTR macros to cast the globals
- * back and from a __FlashStringHelper to get some type-safety. */
+/* Load PROGMEM macro to store global constants in program/flash memory. */
 #include <avr/pgmspace.h>
+
+#include "config.h"
+#include "arduino_secrets.h"
```

Comment:
Not a big fan of splitting up secrets and the rest of the config.h.

SSID is less secret to me than the broker_ip.

Reply:
The macros are, apart from convention, also for the web editor, as explained [here](https://www.andreagrandi.it/2020/12/16/how-to-safely-store-arduino-secrets/) and [here](https://create.arduino.cc/projecthub/Arduino_Genuino/store-your-sensitive-data-safely-when-sharing-a-sketch-e7d0f0). As long as the macro names are rather colission resistant, as with the `SECRET_*` prefix, I think it's rather benign. It has the added bonus that you can put it in PROGMEM, regular mem, both, to your heart's content without changing anything in `arduino_secrets.h`.

I agree that many people will also want to keep their internal network layout, and hence the name of their broker, a secret. I missed this because personally I don't care about hiding my network layout.

SSID being secret is something I personally don't care about, but I can understand where it comes from. Not only can you positively identify where someone lives if you can correlate his SSID as published on GitHub to actually being able to see his SSID broadcast when you are physically near. Also many access points default to having the extension identifier part of the MAC (the bit that is unique) in their SSID. No, I wholly understand people wishing to keep their SSID secret.

> Perhaps place all of config.h in arduino_secrets.h instead then?

Well, except the configuration switches (`OPTIONAL_LIGHT_SENSOR`, `MQTT_TLS`, stuff like that), right? That seems like a good idea.

Reply:
> Well, except the configuration switches

I see how that would short circuit your brain if a `#define` went into `secrets.h` :joy: 

But on the other hand, we'd get a config.h with only a measly 1-3 `#define`s... :thinking: 

Oh well. Either is fine with me.

Reply:
(And ok. Seeing that the SECRET_ stuff is convention, I'm okay with those as macros.)

Reply:
> I see how that would short circuit your brain if a #define went into secrets.h

These digital brains are definitely not short-circuit resistant, no! 😛

> But on the other hand, we'd get a config.h with only a measly 1-3 #defines...

Perhaps we can forego `config.h` and put the configuration defines at the very top of the `.ino`, directly after the documentation.

Reply:
That would work for me as long as I can secretly place them in arduino_secrets.h anyway (without telling you :grin:) and have the same effect :smiling_imp: 

(Otherwise I'd have to work around my custom set defines every time I want to commit something in the ino.)

Reply:
> as long as I can [...] (without telling you)

Apparently you could not, sorry.

😜

I have a slight preference for the user never needing to edit the `.ino` except obviously if they really want to, which means putting config stuff in `config.h`. But you pick what you prefer, at the top of the `.ino` or in a pretty short `config.h`.